src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs

```diff
@@ -142,150 +140,186 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
+                    goto case SyntaxKind.XmlElementStartTag;
+
+                case SyntaxKind.XmlElementStartTag:
+                    var startTagSyntax = (XmlElementStartTagSyntax)node;
+                    nameSyntax = startTagSyntax.Name;
+                    attributes = startTagSyntax.Attributes;
+                    break;
+
+                default:
+                    nameSyntax = null;
+                    attributes = default(SyntaxList<XmlAttributeSyntax>);
+                    break;
             }
 
-            return SpecializedCollections.EmptyEnumerable<CompletionItem>();
+            return nameSyntax?.LocalName.ValueText;
         }
 
-        private IEnumerable<CompletionItem> GetTagsForType(INamedTypeSymbol symbol, DocumentationCommentTriviaSyntax trivia)
+        private bool IsAttributeValueContext(SyntaxToken token, out string tagName, out string attributeName)
         {
-            var items = new List<CompletionItem>();
-
-            var typeParameters = symbol.TypeParameters.Select(p => p.Name).ToSet();
+            XmlAttributeSyntax attributeSyntax = null;
 
-            RemoveExistingTags(trivia, typeParameters, x => AttributeSelector(x, TypeParamTagName));
+            if (token.IsParentKind(SyntaxKind.IdentifierName) && token.Parent.IsParentKind(SyntaxKind.XmlNameAttribute))
+            {
+                // name="bar$$
+                attributeSyntax = (XmlNameAttributeSyntax)token.Parent.Parent;
+            }
+            else if (token.IsKind(SyntaxKind.XmlTextLiteralToken) && token.IsParentKind(SyntaxKind.XmlTextAttribute))
+            {
+                // foo="bar$$
+                attributeSyntax = (XmlTextAttributeSyntax)token.Parent;
+            }
+            else if (token.IsParentKind(SyntaxKind.XmlNameAttribute) || token.IsParentKind(SyntaxKind.XmlTextAttribute))
+            {
+                // name="$$
+                // foo="$$
+                attributeSyntax = (XmlAttributeSyntax)token.Parent;
+                if (token != attributeSyntax.StartQuoteToken)
+                {
+                    attributeSyntax = null;
+                }
+            }
 
-            items.AddRange(typeParameters.Select(t => CreateCompletionItem(FormatParameter(TypeParamTagName, t))));
-            return items;
-        }
 
-        private string AttributeSelector(XmlElementSyntax element, string attribute)
-        {
-            if (!element.StartTag.IsMissing && !element.EndTag.IsMissing)
+            if (attributeSyntax != null)
             {
-                var startTag = element.StartTag;
-                var nameAttribute = startTag.Attributes.OfType<XmlNameAttributeSyntax>().FirstOrDefault(a => a.Name.LocalName.ValueText == NameAttributeName);
-                if (nameAttribute != null)
+                attributeName = attributeSyntax.Name.LocalName.ValueText;
+
+                var emptyElement = attributeSyntax.GetAncestor<XmlEmptyElementSyntax>();
```

Comment:
This is just handling the difference between `<empty-tag/>` and `<start-tag>`. I'll add comments.